### PR TITLE
Order validation

### PR DESCRIPTION
**Pull request by haythem96**

### CODE REVIEW

File: utils/auction.ts [hunk 1]
```diff
@@ -251,3 +253,37 @@ export const verifyMessageWithTime = (data: MessageWithTimeSignature, signature:
...
+  }
+  else if (parseInt(order.quantity) < auction.minSize) {
+    isValidOrder = false;
+    response = 'Order qunatity is less than auction min size'
+  }
+  else if(order.isBuying) {
+    const wethContract = new ethers.Contract(WETH, erc20Abi, provider) as ERC20
+    const traderAllowance = await wethContract.allowance(CRAB_STRATEGY_V2, order.quantity)
+
+    if(traderAllowance < parseInt(order.quantity)) {
```
Comment (KMKoushik) [inline utils/auction.ts:273]: We should also check for balance

**Review by KMKoushik — CHANGES_REQUESTED** [review]
LGTM,

Let's make this PR over beta branch

File: utils/auction.ts [hunk 2]
```diff
@@ -251,3 +253,37 @@ export const verifyMessageWithTime = (data: MessageWithTimeSignature, signature:
   const addr = ethers.utils.verifyTypedData(domain, messageWithTimeType, data, signature!)
   return address.toLowerCase() === addr.toLowerCase()
 }
+
+export const validateOrder = async (order: Order, auction: Auction) => {
+  let isValidOrder = true;
+  let response = '';
+
+  if (order.isBuying != auction.isSelling) {
```
Comment (aleone) [inline utils/auction.ts:261]: does this work ok if the auction details haven't been finalized? (same for min size below)

Comment (haythem96) [reply utils/auction.ts:261]: By finalized u mean auction not started yet/auction details not in the database?

Comment (KMKoushik) [reply utils/auction.ts:260]: Users can submit bids even before the minSize, auction direction is set. If it's not a valid bid manager will omit these bids. 

During this minSize will be 0. So it should by pass this check if the auction is not yet started

File: utils/auction.ts [hunk 3]
```diff
@@ -263,6 +264,46 @@ export const verifyMessageWithTime = (data: MessageWithTimeSignature, signature:
   return address.toLowerCase() === addr.toLowerCase()
 }
 
+export const validateOrder = async (order: Order, auction: Auction) => {
+  let isValidOrder = true;
+  let response = '';
+
+  if (auction.minSize == 0 || auction.auctionEnd == 0) {
```
Comment (KMKoushik) [inline utils/auction.ts:271]: We should not throw this error. If the auction is not started, we allow user create bid which will be filtered by King crab

**Comment by haythem96** [comment]
Tested and it works
Screen Shot 2022-08-26 at 9 07 40 AM

**Pull request merged**